front/src/applications/editor/components/InfraErrors/InfraErrorCorrectorModal.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { FC, useState, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useTranslation } from 'react-i18next';
+
+import { ConfirmModal, useModal } from 'common/BootstrapSNCF/ModalSNCF';
+import { Operation, osrdEditoastApi } from 'common/api/osrdEditoastApi';
+import Loader from 'common/Loader';
+import { saveOperations } from 'reducers/editor';
+import { getInfraID } from '../../../../reducers/osrdconf/selectors';
+
+const InfraErrorCorrectorModal: FC = () => {
+  const { t } = useTranslation();
+  const dispatch = useDispatch();
+  const { closeModal } = useModal();
+
+  const infraId = useSelector(getInfraID);
+  const [isLoading, setIsLoading] = useState(true);
+  const [isSaving, setIsSaving] = useState(false);
+  const [infraAutoFixes, setInfraAutoFixes] = useState<Operation[] | undefined>(undefined);
+
+  const [getInfraAutoFixes] = osrdEditoastApi.endpoints.getInfraByInfraIdAutoFixes.useLazyQuery();
+
+  useEffect(() => {
+    getInfraAutoFixes({ infraId: infraId as number })
+      .unwrap()
+      .then((results) => {
+        setInfraAutoFixes(results);
+        setIsLoading(false);
+      });
+  }, []);
+
+  function displayFixes() {
+    if (infraAutoFixes === undefined || isLoading || isSaving) {
+      return (
+        <Loader
+          msg={t(
+            'Editor.infra-errors.corrector-modal.'.concat(
+              isLoading ? 'loading-loader' : 'saving-loader'
+            )
+          )}
+        />
+      );
+    }
+    if (infraAutoFixes?.length === 0) {
+      return <div>{t('Editor.infra-errors.corrector-modal.total-fixes_zero')}</div>;
+    }
+    const creationCount = infraAutoFixes?.filter(
```

Comment:
infraAutoFixes should be defined since it’s in your condition to return early
```suggestion
    const creationCount = infraAutoFixes.filter(
```